### PR TITLE
Descartes development

### PR DESCRIPTION
**Pull request by tdescartes**
Design first page and add some basic css codes.

### CODE REVIEW

File: portfolio/pom.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
     <!-- Project-specific properties -->
     <exec.mainClass>com.google.sps.ServerMain</exec.mainClass>
-    <googleCloudProjectId>YOUR_PROJECT_ID_HERE</googleCloudProjectId>
+    <googleCloudProjectId>dtuyishime-sps-summer21</googleCloudProjectId>
```
Comment (dgalbraith33) [inline portfolio/pom.xml:19]: FYI: In the future, this sort of change could be it's own PR to keep things separate.

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -1,19 +1,105 @@
...
   <body>
-    <div id="content">
-      <h1>My Portfolio</h1>
-      <p>This is my portfolio.</p>
-      <p>Click <a href="/hello">here</a> to see content created by server code.</p>
-      <p>Click here to get a random message:</p>
-      <button onclick="addRandomGreeting()">Hello</button>
-      <div id="greeting-container"></div>
-    </div>
+      <!-- Main Start-->
```
Comment (dgalbraith33) [inline portfolio/src/main/webapp/index.html:12]: I like how you marked the start and end of the sections. It helps readability a lot!

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -1,19 +1,105 @@
...
+      <div class="main" id="home">
+          <!-- Home section Start-->
+          <section class="home-section align-items-center">
+              <div class="container">
```
Comment (dgalbraith33) [inline portfolio/src/main/webapp/index.html:16]: FYI: the CSS class name container is usually used for the top-level container of the page (i.e. the class you have called "main").

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -1,19 +1,105 @@
...
+                  <div class="row">
+                      <div class="about-img">
+                          <div class="img-box">
+                              <img src="images/fullpicture.jpg" alt="about image">
+                          </div>
+                      </div>
+                      
+                      <div class="about-text">
+
+                          <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
```
Comment (dgalbraith33) [inline portfolio/src/main/webapp/index.html:59]: nit: Don't have lines with more than 80-100 characters if possible.

In HTML you can add a line break without it showing up on the page.

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -1,19 +1,105 @@
...
+
+                          <h3> Skills</h3>
+                          <div class="skill">
+                              <div class="skill-item"> This has html, css, javascript, and others</div>
+
```
Comment (dgalbraith33) [inline portfolio/src/main/webapp/index.html:64]: nit: drop these two blank lines of spacing.

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -1,19 +1,105 @@
...
+                          <div class="img-box">
+                              <img src="images/fullpicture.jpg" alt="about image">
+                          </div>
+                      </div>
+                      
+                      <div class="about-text">
+
+                          <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
+
+                          <h3> Skills</h3>
```
Comment (dgalbraith33) [inline portfolio/src/main/webapp/index.html:61]: nit: drop leading space before Skills

File: portfolio/src/main/webapp/index.html [hunk 7]
```diff
@@ -1,19 +1,105 @@
...
+              </div>
+          </section>
+          <!-- Home section End-->
+      </div>
+      <!--Main End-->
+
+
+      <!-- About section start-->
+      <div class="main" id="about">
+          <section class="about-section sec-padding">
```
Comment (dgalbraith33) [inline portfolio/src/main/webapp/index.html:42]: I would say avoid having CSS classes that aren't defined (here about-section and sec-padding).

If you want to add styles for these in a future PR it is easy enough to add them back. But for now it makes things slightly less clear.

File: portfolio/src/main/webapp/index.html [hunk 8]
```diff
@@ -1,19 +1,105 @@
...
+                              <img src="images/fullpicture.jpg" alt="about image">
+                          </div>
+                      </div>
+                      
+                      <div class="about-text">
+
+                          <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
+
+                          <h3> Skills</h3>
+                          <div class="skill">
```
Comment (dgalbraith33) [inline portfolio/src/main/webapp/index.html:62]: nit: similar thing here with "skill" and "skill-item" classes.

File: portfolio/src/main/webapp/style.css [hunk 9]
```diff
@@ -1,9 +1,161 @@
...
+}
+
+::before,
+::after{
+    box-sizing: border-box ;
+}
+
+body{
+    min-height: 100vh;
+    background-image:linear-gradient(to bottom right, var(--pink-light), var(--cyan-light));
```
Comment (dgalbraith33) [inline portfolio/src/main/webapp/style.css:40]: nit: add a space after "background-image:" to be consistent with the rest of the file.

File: portfolio/src/main/webapp/style.css [hunk 10]
```diff
@@ -1,9 +1,161 @@
...
+    font-size: 16px;
+    font-family: 'Source Code Pro', monospace;
```
Comment (dgalbraith33) [inline portfolio/src/main/webapp/style.css:42]: FYI the way font-family works (apologies if you're already aware of this) is that it tries to use the fonts listed from left to right. So if the "Source Code Pro" font is unavailable for some reason (maybe it can't be loaded). This will switch to a generic monospace font.

Because of this you may want to pick a more specific monospace font for the browser to use in that case. (There is a list here: https://www.w3.org/Style/Examples/007/fonts.en.html). You can test what it will look like if loading the external font fails by simply getting rid of the 'Source Code Pro' temporarily and reloading.

TL;DR in the end you could have something like "font-family: 'Source Code Pro', Courier New, monospace"

**Review by dgalbraith33 — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/index.html (hunk 2 above)
Comment (tdescartes) [reply portfolio/src/main/webapp/index.html:12]: Thank you.
